yugatool/README.md

```diff
@@ -4,21 +4,41 @@
 
 ## Building
 
-#### Requirements
+### Requirements
 
 * make
 * direnv (or source .envrc)
 * golang v1.16+
 * protoc v3.14.0
+* golangci-lint (for `make`)
 
-#### Ubuntu 18.04
+### Install Requirements
+#### -- *_Tested on Ubuntu 18.04_* --
+
+#### `apt` packages ####
 
 ```bash
 apt install direnv make
-snap install protobuf
+snap install protobuf --classic
+```
+
+### Install golang
+
+#### Option 1: snap
+```
 snap install golang
 ```
 
+#### Option 2: manual
+
+https://go.dev/doc/install
+
+
+### Install golangci-lint
```

Comment:
This should be installed automatically as part of `make`. You probably have something misconfigured?

Reply:
This is my first run of `make`. This error persisted until I installed it manually. It's still possible that something is misconfigured, but this is what I ran into.

```
$ make
go install -modfile=.././go.mod github.com/golangci/golangci-lint/cmd/golangci-lint
go: downloading github.com/golangci/golangci-lint v1.42.1
go: downloading github.com/fatih/color v1.12.0
go: downloading github.com/gofrs/flock v0.8.1
go: downloading github.com/pkg/errors v0.9.1
go: downloading github.com/spf13/cobra v1.2.1
go: downloading github.com/spf13/pflag v1.0.5
go: downloading github.com/spf13/viper v1.9.0
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading golang.org/x/tools v0.1.7
go: downloading github.com/go-critic/go-critic v0.5.6
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: downloading github.com/sirupsen/logrus v1.8.1
go: downloading github.com/stretchr/testify v1.7.0
go: downloading github.com/go-xmlfmt/xmlfmt v0.0.0-20191208150333-d5b6f63a941b
go: downloading github.com/golangci/revgrep v0.0.0-20210208091834-cd28932614b5
go: downloading github.com/fsnotify/fsnotify v1.5.1
go: downloading github.com/magiconair/properties v1.8.5
go: downloading github.com/mitchellh/mapstructure v1.4.2
go: downloading github.com/spf13/afero v1.6.0
go: downloading github.com/spf13/cast v1.4.1
go: downloading github.com/spf13/jwalterweatherman v1.1.0
go: downloading github.com/subosito/gotenv v1.2.0
go: downloading gopkg.in/ini.v1 v1.63.2
go: downloading golang.org/x/sys v0.0.0-20211007075335-d3039528d8ac
go: downloading github.com/go-toolsmith/astcast v1.0.0
go: downloading github.com/go-toolsmith/astcopy v1.0.0
go: downloading github.com/go-toolsmith/astequal v1.0.0
go: downloading github.com/go-toolsmith/astfmt v1.0.0
go: downloading github.com/go-toolsmith/astp v1.0.0
go: downloading github.com/go-toolsmith/strparse v1.0.0
go: downloading github.com/go-toolsmith/typep v1.0.2
go: downloading github.com/quasilyte/go-ruleguard v0.3.4
go: downloading github.com/quasilyte/regex/syntax v0.0.0-20200407221936-30656e2c4a95
go: downloading 4d63.com/gochecknoglobals v0.0.0-20201008074935-acfc0b28355a
go: downloading github.com/Antonboom/errname v0.1.4
go: downloading github.com/BurntSushi/toml v0.4.1
go: downloading github.com/Djarvur/go-err113 v0.0.0-20210108212216-aea10b59be24
go: downloading github.com/OpenPeeDeeP/depguard v1.0.1
go: downloading github.com/alexkohler/prealloc v1.0.0
go: downloading github.com/ashanbrown/forbidigo v1.2.0
go: downloading github.com/ashanbrown/makezero v0.0.0-20210520155254-b6261585ddde
go: downloading github.com/bkielbasa/cyclop v1.2.0
go: downloading github.com/bombsimon/wsl/v3 v3.3.0
go: downloading github.com/charithe/durationcheck v0.0.8
go: downloading github.com/daixiang0/gci v0.2.9
go: downloading github.com/denis-tingajkin/go-header v0.4.2
go: downloading github.com/esimonov/ifshort v1.0.2
go: downloading github.com/fzipp/gocyclo v0.3.1
go: downloading github.com/golangci/check v0.0.0-20180506172741-cfe4005ccda2
go: downloading github.com/golangci/dupl v0.0.0-20180902072040-3e9179ac440a
go: downloading github.com/golangci/go-misc v0.0.0-20180628070357-927a3d87b613
go: downloading github.com/golangci/gofmt v0.0.0-20190930125516-244bba706f1a
go: downloading github.com/golangci/lint-1 v0.0.0-20191013205115-297bf364a8e0
go: downloading github.com/golangci/maligned v0.0.0-20180506175553-b1d89398deca
go: downloading github.com/golangci/misspell v0.3.5
go: downloading github.com/golangci/unconvert v0.0.0-20180507085042-28b1c447d1f4
go: downloading github.com/gordonklaus/ineffassign v0.0.0-20210225214923-2e10b2664254
go: downloading github.com/gostaticanalysis/forcetypeassert v0.0.0-20200621232751-01d4955beaa5
go: downloading github.com/gostaticanalysis/nilerr v0.1.1
go: downloading github.com/jgautheron/goconst v1.5.1
go: downloading github.com/jingyugao/rowserrcheck v1.1.0
go: downloading github.com/jirfag/go-printf-func-name v0.0.0-20200119135958-7558a9eaa5af
go: downloading github.com/julz/importas v0.0.0-20210419104244-841f0c0fe66d
go: downloading github.com/kisielk/errcheck v1.6.0
go: downloading github.com/kulti/thelper v0.4.0
go: downloading github.com/kunwardeep/paralleltest v1.0.2
go: downloading github.com/kyoh86/exportloopref v0.1.8
go: downloading github.com/ldez/gomoddirectives v0.2.2
go: downloading github.com/ldez/tagliatelle v0.2.0
go: downloading github.com/maratori/testpackage v1.0.1
go: downloading github.com/matoous/godox v0.0.0-20210227103229-6504466cf951
go: downloading github.com/mbilski/exhaustivestruct v1.2.0
go: downloading github.com/mgechev/dots v0.0.0-20190921121421-c36f7dcfbb81
go: downloading github.com/mgechev/revive v1.1.1
go: downloading github.com/moricho/tparallel v0.2.1
go: downloading github.com/nakabonne/nestif v0.3.0
go: downloading github.com/nishanths/exhaustive v0.2.3
go: downloading github.com/nishanths/predeclared v0.2.1
go: downloading github.com/polyfloyd/go-errorlint v0.0.0-20210722154253-910bb7978349
go: downloading github.com/ryancurrah/gomodguard v1.2.3
go: downloading github.com/ryanrolds/sqlclosecheck v0.3.0
go: downloading github.com/sanposhiho/wastedassign/v2 v2.0.6
go: downloading github.com/securego/gosec/v2 v2.8.1
go: downloading github.com/shazow/go-diff v0.0.0-20160112020656-b6b7b6733b8c
go: downloading github.com/sonatard/noctx v0.0.1
go: downloading github.com/sourcegraph/go-diff v0.6.1
go: downloading github.com/ssgreg/nlreturn/v2 v2.1.0
go: downloading github.com/tdakkota/asciicheck v0.0.0-20200416200610-e657995f937b
go: downloading github.com/tetafro/godot v1.4.9
go: downloading github.com/timakin/bodyclose v0.0.0-20200424151742-cb6215831a94
go: downloading github.com/tomarrell/wrapcheck/v2 v2.3.0
go: downloading github.com/tommy-muehle/go-mnd/v2 v2.4.0
go: downloading github.com/ultraware/funlen v0.0.3
go: downloading github.com/ultraware/whitespace v0.0.4
go: downloading github.com/uudashr/gocognit v1.0.5
go: downloading github.com/yeya24/promlinter v0.1.0
go: downloading honnef.co/go/tools v0.2.1
go: downloading mvdan.cc/gofumpt v0.1.1
go: downloading mvdan.cc/interfacer v0.0.0-20180901003855-c20040233aed
go: downloading mvdan.cc/unparam v0.0.0-20210104141923-aac4ce9116a7
go: downloading github.com/hashicorp/go-multierror v1.1.1
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/pmezard/go-difflib v1.0.0
go: downloading github.com/stretchr/objx v0.2.0
go: downloading golang.org/x/text v0.3.7
go: downloading github.com/hashicorp/hcl v1.0.0
go: downloading github.com/pelletier/go-toml v1.9.4
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading github.com/gobwas/glob v0.2.3
go: downloading github.com/kisielk/gotool v1.0.0
go: downloading github.com/gostaticanalysis/comment v1.4.1
go: downloading github.com/gostaticanalysis/analysisutil v0.4.1
go: downloading golang.org/x/mod v0.5.1
go: downloading github.com/ettle/strcase v0.1.1
go: downloading github.com/fatih/structtag v1.2.0
go: downloading github.com/Masterminds/semver v1.5.0
go: downloading github.com/phayes/checkstyle v0.0.0-20170904204023-bfd46e6a821d
go: downloading github.com/nbutton23/zxcvbn-go v0.0.0-20210217022336-fa2cb2858354
go: downloading github.com/prometheus/client_golang v1.7.1
go: downloading github.com/prometheus/client_model v0.2.0
go: downloading mvdan.cc/lint v0.0.0-20170908181259-adc824a0674b
go: downloading github.com/google/go-cmp v0.5.6
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/chavacava/garif v0.0.0-20210405164556-e8a0a408d6af
go: downloading github.com/olekukonko/tablewriter v0.0.5
go: downloading github.com/beorn7/perks v1.0.1
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/cespare/xxhash v1.1.0
go: downloading github.com/prometheus/common v0.10.0
go: downloading github.com/prometheus/procfs v0.6.0
go: downloading github.com/mattn/go-runewidth v0.0.12
go: downloading google.golang.org/protobuf v1.27.1
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/rivo/uniseg v0.1.0
go install -modfile=.././go.mod github.com/onsi/ginkgo/ginkgo
go: downloading github.com/onsi/ginkgo v1.16.4
go: downloading github.com/go-task/slim-sprig v0.0.0-20210107165309-348f09dbbbc0
go: downloading github.com/nxadm/tail v1.4.8
go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go install -modfile=.././go.mod github.com/spf13/cobra/cobra
go install -modfile=.././go.mod github.com/go-swagger/go-swagger/cmd/swagger
go: downloading github.com/go-swagger/go-swagger v0.28.0
go: downloading github.com/jessevdk/go-flags v1.5.0
go: downloading github.com/go-openapi/analysis v0.20.1
go: downloading github.com/go-openapi/loads v0.20.3
go: downloading github.com/go-openapi/runtime v0.20.0
go: downloading github.com/go-openapi/spec v0.20.4
go: downloading github.com/go-openapi/strfmt v0.20.3
go: downloading github.com/go-openapi/swag v0.19.15
go: downloading github.com/go-openapi/validate v0.20.3
go: downloading github.com/gorilla/handlers v1.5.1
go: downloading github.com/toqueteos/webbrowser v1.2.0
go: downloading github.com/go-openapi/jsonpointer v0.19.5
go: downloading github.com/go-openapi/errors v0.20.1
go: downloading github.com/asaskevich/govalidator v0.0.0-20210307081110-f21760c49a8d
go: downloading github.com/oklog/ulid v1.3.1
go: downloading go.mongodb.org/mongo-driver v1.7.3
go: downloading github.com/go-openapi/jsonreference v0.19.6
go: downloading github.com/mailru/easyjson v0.7.7
go: downloading github.com/go-openapi/inflect v0.19.0
go: downloading github.com/kr/pretty v0.3.0
go: downloading github.com/felixge/httpsnoop v1.0.2
go: downloading github.com/PuerkitoBio/purell v1.1.1
go: downloading github.com/josharian/intern v1.0.0
go: downloading github.com/kr/text v0.2.0
go: downloading github.com/rogpeppe/go-internal v1.8.0
go: downloading github.com/go-stack/stack v1.8.1
go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
go: downloading golang.org/x/net v0.0.0-20211008194852-3b03d305991f
go install google.golang.org/protobuf/cmd/protoc-gen-go
go fmt ./...
go: downloading github.com/go-logr/logr v1.1.0
go: downloading github.com/icza/gox v0.0.0-20210726201659-cd40a3f8d324
go: downloading github.com/yugabyte/gocql v0.0.0-20200602185649-ef3952a45ff4
go: downloading github.com/blang/vfs v1.0.0
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/yudai/gojsondiff v1.0.0
go: downloading github.com/go-logr/zapr v1.1.0
go: downloading go.uber.org/zap v1.19.0
go: downloading github.com/onsi/gomega v1.16.0
go: downloading github.com/alexeyco/simpletable v1.0.0
go: downloading github.com/ghodss/yaml v1.0.0
go: downloading github.com/hako/durafmt v0.0.0-20210608085754-5c1018a4e16b
go: downloading github.com/spyzhov/ajson v0.4.2
go: downloading github.com/golang/snappy v0.0.3
go: downloading github.com/hailocab/go-hostpool v0.0.0-20160125115350-e80d13ce29ed
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading github.com/sergi/go-diff v1.1.0
go: downloading github.com/yudai/golcs v0.0.0-20170316035057-ecda9a501e82
go: downloading go.uber.org/atomic v1.7.0
go: downloading go.uber.org/multierr v1.6.0
go vet ./...
golangci-lint run
make: golangci-lint: Command not found
make: *** [../global.mk:23: lint] Error 127
```

Reply:
It looks like you didn't properly set up `direnv`, or didn't source the file... We should add a new section to the README about direnv, and also make building fail when the GOPATH doesn't point to `yb-tools/bin/`